### PR TITLE
Update Setup Documentation and Dev Dependencies

### PR DESCRIPTION
**Pull request by kry0sc0pic**
Added instructions on how to setup bot for developers using `poetry` and included some common commands needed during development using `poetry`.

Updated the dev dependencies to add missing package `pyright`. This will be needed for #30

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,21 @@ To use this bot, you need to do the following:
...
+5. Install `poetry` using `pip install poetry`
+6. Install the required python packages by running `poetry install`
+7. Run the `main.py` file to start the bot.
+
+### Commonly used poetry commands
+- **Adding Dependencies**
+`poetry add <package_name>`
+- **Updating requirements files**
+`poetry export --without-hashes -o requirements.txt`
+`poetrt export --without-hashes -o dev-requirements.txt --with dev`
```
Comment (lmaotrigine) [inline README.md]:
```suggestion
`poetry export --without-hash --only dev -o dev-requirements.txt`
```

File: README.md [hunk 2]
```diff
@@ -14,6 +14,21 @@ To use this bot, you need to do the following:
...
+4. Rename `.env.example` to `.env`
+5. Install `poetry` using `pip install poetry`
+6. Install the required python packages by running `poetry install`
+7. Run the `main.py` file to start the bot.
+
+### Commonly used poetry commands
+- **Adding Dependencies**
+`poetry add <package_name>`
+- **Updating requirements files**
+`poetry export --without-hashes -o requirements.txt`
```
Comment (lmaotrigine) [inline README.md]:
```suggestion
`poetry export --without-hashes --only main -o requirements.txt`
```

File: pyproject.toml [hunk 3]
```diff
@@ -18,6 +18,7 @@ python-dotenv = "^1.0.0"
 black = "^23.3.0"
 isort = "^5.12.0"
 ipython = "^8.14.0"
+pyright = "^1.1.318"
```
Comment (lmaotrigine) [inline pyproject.toml]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -14,6 +14,21 @@ To use this bot, you need to do the following:
 5. Install the required Python packages by running `pip install -r requirements.txt.`
 6. Run the `main.py` file to start the bot.
 
+## Setup for Development
+1. Clone this repository to your local machine.
+2. Modify `config.py` with your settings.
+3. Add the required secrets to `.env.example`
+4. Rename `.env.example` to `.env`
+5. Install `poetry` using `pip install poetry`
```
Comment (lmaotrigine) [inline README.md]: don't recommend pip installation. The official [docs](https://python-poetry.org/docs/#installation) state to use the official installer unless you know what you're doing.

**Review by lmaotrigine — CHANGES_REQUESTED** [review]
First pass, if something is unclear, I'm happy to clarify.

File: README.md [hunk 5]
```diff
@@ -24,11 +24,10 @@ To use this bot, you need to do the following:
...
 
 ### Commonly used poetry commands
-- **Adding Dependencies**
-`poetry add <package_name>`
-- **Updating requirements files**
-`poetry export --without-hashes -o requirements.txt`
-`poetrt export --without-hashes -o dev-requirements.txt --with dev`
+- **Adding Dependency** `poetry add <package_name>`
+- **Adding Dev Dependency** `poetry add <package_name> --group dev`
+- **Updating Requirements File** `poetry export --without-hashes -o requirements.txt`
```
Comment (lmaotrigine) [inline README.md]:
```suggestion
- **Updating Requirements File** `poetry export --without-hashes --only main -o requirements.txt`
```

File: README.md [hunk 6]
```diff
@@ -24,11 +24,10 @@ To use this bot, you need to do the following:
...
+- **Updating Dev Requirements File** `poetry export --without-hashes -o dev-requirements.txt --with dev`
```
Comment (lmaotrigine) [inline README.md]:
```suggestion
- **Updating Dev Requirements File** `poetry export --without-hashes -o dev-requirements.txt --only dev`

**Review by lmaotrigine — COMMENTED** [review]
These exports are still incorrect

File: README.md (hunk 4 above)
Comment (kry0sc0pic) [reply README.md]: Ok will update

File: README.md [hunk 7]
```diff
@@ -19,15 +19,19 @@ To use this bot, you need to do the following:
...
-### Commonly used poetry commands
+### Commonly used development commands
 - **Adding Dependency** `poetry add <package_name>`
 - **Adding Dev Dependency** `poetry add <package_name> --group dev`
-- **Updating Requirements File** `poetry export --without-hashes -o requirements.txt`
-- **Updating Dev Requirements File** `poetry export --without-hashes -o dev-requirements.txt --with dev`
+- **Updating Requirements File** `poetry export --without-hashes --only main -o requirements.txt`
+- **Updating Dev Requirements File** `poetry export --without-hashes -o dev-requirements.txt --only dev`
+- **Sorting Imports** `isort .`
+- **Formatting Code** `black .`
```
Comment (lmaotrigine) [inline README.md:33]: maybe mention running `pyright` as well, since not everyone may use an editor/IDE that supports it.

**Review by lmaotrigine — CHANGES_REQUESTED** [review]
Also, the requirements files are now out of date. Please run the export commands and push the changes.

**Review by lmaotrigine — APPROVED** [review]

**Pull request merged**